exporter/elasticsearchexporter/model.go

```diff
@@ -137,3 +208,109 @@ func scopeToAttributes(scope pcommon.InstrumentationScope) pcommon.Map {
 	}
 	return attrs
 }
+
+func encodeLogAttributesECSMode(document *objmodel.Document, attrs pcommon.Map, conversionMap map[string]string) {
+	if len(conversionMap) == 0 {
+		// No conversions to be done; add all attributes at top level of
+		// document.
+		document.AddAttributes("", attrs)
+		return
+	}
+
+	attrs.Range(func(k string, v pcommon.Value) bool {
+		// If ECS key is found for current k in conversion map, use it.
+		if ecsKey, exists := conversionMap[k]; exists {
+			if ecsKey == "" {
+				// Skip the conversion for this k.
+				return true
+			}
+
+			document.AddAttribute(ecsKey, v)
+			return true
+		}
+
+		// Otherwise, add key at top level with attribute name as-is.
+		document.AddAttribute(k, v)
+		return true
+	})
+}
+
+func encodeLogAgentNameECSMode(document *objmodel.Document, resource pcommon.Resource) {
+	// Parse out telemetry SDK name, language, and distro name from resource
+	// attributes, setting defaults as needed.
+	telemetrySdkName := "otlp"
+	var telemetrySdkLanguage, telemetryDistroName string
+
+	attrs := resource.Attributes()
+	if v, exists := attrs.Get(semconv.AttributeTelemetrySDKName); exists {
+		telemetrySdkName = v.Str()
+	}
+	if v, exists := attrs.Get(semconv.AttributeTelemetrySDKLanguage); exists {
+		telemetrySdkLanguage = v.Str()
+	}
+	if v, exists := attrs.Get(semconv.AttributeTelemetryDistroName); exists {
+		telemetryDistroName = v.Str()
+		if telemetrySdkLanguage == "" {
+			telemetrySdkLanguage = "unknown"
+		}
+	}
+
+	// Construct agent name from telemetry SDK name, language, and distro name.
+	agentName := telemetrySdkName
+	if telemetryDistroName != "" {
+		agentName = fmt.Sprintf("%s/%s/%s", agentName, telemetrySdkLanguage, telemetryDistroName)
+	} else if telemetrySdkLanguage != "" {
+		agentName = fmt.Sprintf("%s/%s", agentName, telemetrySdkLanguage)
+	}
+
+	// Set agent name in document.
+	document.AddString("agent.name", agentName)
+
+	// Set service language name in document.
+	if telemetrySdkLanguage == "" {
+		telemetrySdkLanguage = "unknown"
+	}
+	document.AddString("service.language.name", telemetrySdkLanguage)
```

Comment:
> One thing I think we should not carry across from apm-data is setting service.language.name when telemetry.sdk.language is unspecified. 

Sounds good. But maybe we have to adapt some of the assumptions we have in the APM UI about this field always being present.

> Maybe we can just skip setting service.language.name altogether for now, since it's included in agent.name?

The `service.language.name` attribute is used a lot in our UI. Since it directly translates to `service.language.name` I think I'd just add it to the `resourceAttrsConversionMap`.

Reply:
> The service.language.name attribute is used a lot in our UI. Since it directly translates to service.language.name I think I'd just add it to the resourceAttrsConversionMap.

This PR is focused on logs though, and I think the `service.language.name` usage in the (APM) UI is not relevant here. Could we start without it, and add it in later if there's a need?

Reply:
let's do that

Reply:
Removed code for setting `service.language.name` in cb5a3b88bb.